crypto/src/engine/blst/g1.rs

```diff
@@ -0,0 +1,115 @@
+use std::mem::MaybeUninit;
+
+use blst::{
+    blst_p1, blst_p1_affine, blst_p1_affine_compress, blst_p1_affine_in_g1, blst_p1_from_affine,
+    blst_p1_mult, blst_p1_to_affine, blst_p1_uncompress, blst_p1s_to_affine,
+};
+use rayon::prelude::{IntoParallelIterator, ParallelIterator};
+
+use crate::{ParseError, G1};
+
+use super::{scalar::ScalarBlst, BLSTAlgebra};
+
+pub struct G1BlstAffine(MaybeUninit<blst_p1_affine>);
+pub struct G1BlstProjective(MaybeUninit<blst_p1>);
+
+impl TryFrom<G1> for G1BlstAffine {
+    type Error = ParseError;
+
+    fn try_from(g1: G1) -> Result<Self, Self::Error> {
+        let mut p = std::mem::MaybeUninit::<blst_p1_affine>::zeroed();
+        unsafe {
+            blst_p1_uncompress(p.as_mut_ptr(), g1.0.as_ptr());
+        }
+
+        Ok(Self(p))
+    }
+}
+
+impl TryFrom<G1BlstAffine> for G1 {
```

Comment:
This can be regular `From`. It should never fail.

Reply:
yes good point! still a bit of copy waste going on, will make all the unnecessary ones regular `From`s